### PR TITLE
Use go integration tests

### LINKED ISSUE #54: Do not use current directory as temp directory
Some files are temporarily placed in the current directory.

### PR DESCRIPTION
**Pull request by r2d4**
We can run these in parallel once we fix #54 

Changed the travis configuration to 
 - not use sudo, since it is no longer required.
 - use go 1.8.3
 - run `make test integration` instead of bash script.

### CODE REVIEW

**Comment by r2d4** [comment]
Fixes #50

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
-sudo: required
-dist: trusty
-
 language: go
+os: linux
```
Comment (nkubala) [inline .travis.yml]: I think you can specify multiple OSes here, should we run these tests on all platforms?

```
os:
  - linux
  - osx
  - windows
```

File: tests/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+const (
+	diffBase     = "gcr.io/gcp-runtimes/diff-base"
+	diffModified = "gcr.io/gcp-runtimes/diff-modified"
+
+	aptBase     = "gcr.io/gcp-runtimes/apt-base"
+	aptModified = "gcr.io/gcp-runtimes/apt-modified"
+
+	// Why is this node-modified:2.0?
+	nodeBase     = "gcr.io/gcp-runtimes/node-modified:2.0"
```
Comment (nkubala) [inline tests/integration_test.go]: hmm...I wonder if the source code for these images is lying around anywhere, I'm not a huge fan of this being abstracted

Comment (r2d4) [reply tests/integration_test.go]: Yeah as a TODO:  get rid of these + get rid of the `_expected.json` files.

File: .travis.yml (hunk 1 above)
Comment (r2d4) [reply .travis.yml]: I tried this, but travis always has a backlog of OSX jobs (its at ~800 now).  I think its probably best to just test on one platform for now.

**Pull request merged**